src/lib/stark/ledger/createSignedOrder.js

```diff
@@ -0,0 +1,102 @@
+const Eth = require('@ledgerhq/hw-app-eth').default
+const byContractAddress = require('@ledgerhq/hw-app-eth/erc20')
+  .byContractAddress
+const DVFError = require('../../dvf/DVFError')
+const BN = require('bignumber.js')
+const _ = require('lodash')
+const selectTransport = require('../../ledger/selectTransport')
+
+module.exports = async (dvf, path, starkOrder) => {
+  const Transport = selectTransport(dvf.isBrowser)
+
+  const buyCurrency = _.find(dvf.config.tokenRegistry, {
+    starkTokenId: starkOrder.tokenBuy
+  })
+  const sellCurrency = _.find(dvf.config.tokenRegistry, {
+    starkTokenId: starkOrder.tokenSell
+  })
+  console.log(buyCurrency, sellCurrency)
+  const transport = await Transport.create()
+  const eth = new Eth(transport)
+  const tempKey = (await eth.starkGetPublicKey(path)).toString('hex')
+  const starkPublicKey = {
+    x: tempKey.substr(2, 64),
+    y: tempKey.substr(66)
+  }
+  console.log(tempKey, starkPublicKey)
+
+  // TODO Extract below code to a utility method
+  // to be used for both buy as sell tokens and
+  // for transfer method as well as well
+
+  let buyTokenAddress = buyCurrency.tokenAddress
+
+  if (buyTokenAddress) {
+    const buyTokenInfo = byContractAddress(buyTokenAddress)
+    buyTokenAddress = buyTokenAddress.substr(2)
+    if (buyTokenInfo) {
+      await eth.provideERC20TokenInformation(buyTokenInfo)
+    } else {
+      if (process.env.NODE_ENV === 'test') {
+        let tokenInfo = {}
+        tokenInfo['data'] = Buffer.from(
+          `00${buyTokenAddress}0000000000000000`,
+          'hex'
+        )
+        await eth.provideERC20TokenInformation(tokenInfo)
+      } else {
```

Comment:
same as in the other comment
if you need `buyTokenInfo` to be defined in tests, you should mock `byContractAddress` instead of hardcoding behaviour in the application code

Reply:
Same as above.

Further, take a look at the transport mocking mechanism here: https://github.com/btchip/ledgerjs/blob/starkware/packages/hw-app-eth/tests/Eth.test.js It wont help us separate test-specific code from application code.

Reply:
interesting, if the mock you have here now works for you, you could probably just use [jest's mocking functionality](https://jestjs.io/docs/en/mock-functions.html#mock-implementations) in order to return the desired `tokenInfo` from `byContractAddress`